File: backend/components/metiers/automobile/utils/metrics.py
"""
metrics.py – Collecte et reporting des métriques environnementales (Python)
"""
from collections import defaultdict

_metrics = defaultdict(list)

def record_metric(name, value):
    _metrics[name].append(value)

def get_average_metric(name):
    arr = _metrics[name]
    if not arr:
        return None
    return sum(arr) / len(arr)

def get_all_metrics():
    return dict(_metrics)

File: backend/components/metiers/securite/utils/metrics.js
// metrics.js – Collecte et reporting des métriques environnementales (Node.js)
const metrics = {};

module.exports = {
  record: (name, value) => {
    metrics[name] = (metrics[name] || []).concat([value]);
  },
  getAverage: (name) => {
    const arr = metrics[name] || [];
    if (!arr.length) return null;
    return arr.reduce((a, b) => a + b, 0) / arr.length;
  },
  getAll: () => metrics
};
